modules/tinymce/src/core/main/ts/fmt/CaretFormat.ts

```diff
@@ -289,7 +299,12 @@ const removeCaretFormat = function (editor: Editor, name: string, vars: FormatVa
     }
 
     removeCaretContainerNode(editor, caretContainer, false);
-    selection.setCursorLocation(caretNode, 1);
+
+    if (uniqueFormats.length > 0) {
+      editor.formatter.remove(name, vars, newCaretContainer, similar);
```

Comment:
Hmm, removing this here could cause the text node to be recreated, the zwsp to be trimmed or all sorts of potential issues. Is there a reason this can't be done before calling `insertFormatNodesIntoCaretContainer`?

Reply:
Ahh, actually it's probably because adding `formatNode` to the `parents` adds it back. I'll see if I can think of a way around that...

Reply:
I wonder, could we just remove the format from the `formatNode`?

Reply:
I am pretty sure I tried that and it didn't work properly. Although, I will check again to be sure.

Reply:
This should be fixed now. Thanks @lnewson for finding a solution!